### PR TITLE
fix: Fixes for the HTTP transport

### PR DESCRIPTION
**Pull request by bruno-garcia**
* DSN has to be parsed and Sentry URL built
* Timeouts are in millseconds

### CODE REVIEW

File: sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java [hunk 1]
```diff
@@ -167,6 +150,21 @@ public TransportResult send(SentryEvent event) throws IOException {
     }
   }
 
+  private void LogErrorInPayload(HttpURLConnection connection) {
```
Comment (marandaneto) [inline sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java]: camel case

File: sentry-core/src/main/java/io/sentry/core/EnvelopeReader.java [hunk 2]
```diff
@@ -36,6 +36,7 @@
     while ((currentLength = stream.read(buffer)) > 0) {
       for (int i = 0; envelopeEndHeaderOffset == -1 && i < currentLength; i++) {
         if (buffer[i] == '\n') {
+          // TODO: IDE warns this is always true, is it?
```
Comment (marandaneto) [inline sentry-core/src/main/java/io/sentry/core/EnvelopeReader.java]: @bruno-garcia can you have a look at it?

File: sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java [hunk 3]
```diff
@@ -49,34 +40,33 @@
...
-    this.sentryUrl = new URI(options.getDsn()).toURL();
     this.bypassSecurity = bypassSecurity;
+    this.sentryUrl = sentryUrl;
   }
 
   // giving up on testing this method is probably the simplest way of having the rest of the class
   // testable...
   @VisibleForTesting
   protected HttpURLConnection open(URL url, Proxy proxy) throws IOException {
+    // why do we need url here? its not used
```
Comment (marandaneto) [inline sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java]: @metlos should we remove `url`  as we use only `sentryUrl`?

File: sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java [hunk 4]
```diff
@@ -89,6 +79,12 @@ public TransportResult send(SentryEvent event) throws IOException {
     connection.setRequestMethod("POST");
     connection.setDoOutput(true);
     connection.setRequestProperty("Content-Encoding", "UTF-8");
+    connection.setRequestProperty("Content-Type", "application/json");
+    connection.setRequestProperty("Accept", "application/json");
```
Comment (marandaneto) [inline sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java]: @bruno-garcia it works without that, but it's very common to set anyway

File: sentry-core/src/main/java/io/sentry/core/EnvelopeReader.java (hunk 2 above)
Comment (bruno-garcia) [reply sentry-core/src/main/java/io/sentry/core/EnvelopeReader.java]: would be a different PR eh?

Comment (bruno-garcia) [reply sentry-core/src/main/java/io/sentry/core/EnvelopeReader.java]: by eye I can't see how. buffer gets modified on each call to `read`

File: sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java [hunk 5]
```diff
@@ -1,25 +1,16 @@
 package io.sentry.core.transport;
 
 import static io.sentry.core.ILogger.log;
-import static io.sentry.core.SentryLevel.DEBUG;
```
Comment (bruno-garcia) [inline sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java]: I think we need to bring these imports back so spotless let's us thru

File: sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java (hunk 4 above)
Comment (bruno-garcia) [reply sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java]: I don't set in .NET, this prob came from `sentry-java`.
No opinion here, could probably get away without it as u suggest.
@HazAT thoughts?

File: sentry-core/src/main/java/io/sentry/core/EnvelopeReader.java (hunk 2 above)
Comment (marandaneto) [reply sentry-core/src/main/java/io/sentry/core/EnvelopeReader.java]: I really didn't check it, just got the warning from the IDE

File: sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java (hunk 5 above)
Comment (marandaneto) [reply sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java]: spotless is good here, is CI failing?

**Pull request merged**